plugin.video.tvvn/resources/settings.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
+<settings>
+	<category label="30000">
+		<setting id="descriptions" type="bool" label="30001" default="true"/>
+		<setting id="descriptions_on_right" type="bool" label="30002" default="true"/>
+		<setting type="sep"/>
+		<setting id="json_url_auto_update" type="bool" label="30003" default="true"/>
+		<setting id="json_url" type="text" label="30004" default='http://zecoj.com/xbmc/data.json.gz' />
```

Comment:
Sorry, not allowed. You're required to parse the sources in Kodi or point to the official stream URLs. We don't allow shipping arbitrary files from personal servers

Reply:
Hmm this was accepted to Leia and older releases previously, the only change for Matrix was Python 3 changes. 

This is to deploy quick fixes in case provider streams are broken. In most cases, the fix simply is change to string matching patterns to get the right URL. Is there a better way to do this as this really does not change the code of the plugin and making a pull request every time a URL is broken feels a bit excessive.

Can you please point me to the rules of what we are allowed/not allowed to do? I'd like to be able to offer quick turnaround fixes for users so would love to explore some other options.

Reply:
@zecoj we can't allow stream url from a personal server as we are the ones being responsible (and liable) to make sure that all streams are 100% legal.

theoretically nothing could stop you from added illegal/piracy streams to the list once this addon is merged.

if this was excepted in Leia, than it's obviously our bad and it slipped through the cracks.

other do's and don'ts are listed here: https://kodi.wiki/view/Add-on_rules